src/Config.hs

```diff
@@ -332,3 +335,36 @@ readValueFromEnvWithDefault :: (Read a, OptParse.HasValue f)
 readValueFromEnvWithDefault key defVal envVars
   =  value defVal
   <> readValueFromEnv key envVars
+
+-- | Search for environment files in default locations and load them in order.
+--
+-- This function tries to read the following files in order to obtain
+-- environment configuration. This is implicit behavior and allows the user to
+-- configure vaultenv without setting up environment variables or passing CLI
+-- flags. This is nicer for interactive usage.
+readConfigFromEnvFiles :: IO [(String, String)]
+readConfigFromEnvFiles = do
+  homeDir <- Dir.getHomeDirectory
```

Comment:
What happens if the current user doesn't have a home directory?

Reply:
That can indeed fail, (I expected this to just read `$HOME`, but I guess that variable can also be unset and this function also does an existence test).

Changed this to `getXdgDir` and am now catching errors here